misc/dbt-materialize/dbt/include/materialize/macros/deploy/deploy_init.sql

```diff
@@ -181,7 +183,140 @@
             {% endset %}
             {{ run_query(create_cluster) }}
             {{ set_cluster_ci_tag() }}
+            {{ internal_copy_cluster_grants(schema, deploy_schema) }}
         {% endif %}
     {% endif %}
 {% endfor %}
 {% endmacro %}
+
+{% macro internal_copy_schema_default_privs(from, to) %}
+{% set revoke_initial_defaults %}
+SELECT
+  'ALTER DEFAULT PRIVILEGES ' ||
+  CASE
+    WHEN object_owner = 'PUBLIC' THEN 'FOR ALL ROLES '
+    ELSE 'FOR ROLE ' || object_owner
+  END ||
+  'IN SCHEMA quote_ident({{ to }}) ' ||
+  'REVOKE '    || privilege_type || ' ' ||
+  'ON ' || object_type || 's ' ||
+  'TO ' || grantee || ';'
+FROM mz_internal.mz_show_default_privileges
+WHERE database = current_database() AND schema = lower(trim('{{ from }}'))
```

Comment:
I realize the `lower(trim('{{ schema }'))` pattern is something that you've been using elsewhere, but I don't understand it and it seems a little sketchy to me! What's up with `lower` and `trim`? 

Also, I think you want to be using [`dbt.string_literal`](https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros#string_literal) rather than `'{{ from }}'` to safely handle schema names with single quotes within.

Put another way, what goes wrong with a condition like this?

```
schema = {{ dbt.string_literal(from) }}
```

Reply:
It's meant to normalize user input that may have extra whitespace. The lower is for pg normalization but quote_ident is better. I'll update this.

Reply:
Yeah, spaces and uppercase letters are valid idents so I don’t think you want to apply trimming and lowering. Also careful, quote_ident doesn’t seem like what you want here, as you want a string literal, not an ident. I think dbt.string_literal is the thing you want.